.cirrus.yml

```diff
@@ -22,584 +40,532 @@ task:
+      script:
         - dart --enable-asserts dev\bots\test.dart
+
+    # TODO(ianh): name: add_to_app_tests-windows
+
     - name: customer_testing-windows
-      test_script:
+      script:
         - CMD /S /C "IF EXIST "bin\cache\pkg\tests\" RMDIR /S /Q bin\cache\pkg\tests"
         - git clone https://github.com/flutter/tests.git bin\cache\pkg\tests
         - dart --enable-asserts dev\customer_testing\run_tests.dart --skip-on-fetch-failure --skip-template bin/cache/pkg/tests/registry/*.test
-    - name: integration_tests_gradle1-windows
-      env:
-        SHARD: integration_tests
-        SUBSHARD: gradle1
-      test_script:
-        - dart --enable-asserts dev\bots\test.dart
-      container:
-        cpu: 4
-        memory: 12G
-    - name: integration_tests_gradle2-windows
-      env:
-        SHARD: integration_tests
-        SUBSHARD: gradle2
-      test_script:
-        - dart --enable-asserts dev\bots\test.dart
-      container:
-        cpu: 4
-        memory: 12G
 
+# MACOS SHARDS
+# Mac doesn't use caches because they apparently take longer to populate and save
+# than just fetching the data in the first place.
 task:
-  use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true'
-  name: deploy_gallery-macos
-  depends_on:
-    - analyze
-  env:
-    # Name the SDK directory to include a space so that we constantly
-    # test path names with spaces in them.
-    CIRRUS_WORKING_DIR: "/tmp/flutter sdk"
-    SHARD: deploy_gallery
-    # Apple Fastlane password.
-    FASTLANE_PASSWORD: ENCRYPTED[4b1f0b8d52874e9de965acd46c79743f3b81f3a513614179b9be7cf53dc8258753e257bdadb11a298ee455259df21865]
-    # Private repo for publishing certificates.
-    PUBLISHING_MATCH_CERTIFICATE_REPO: ENCRYPTED[3c0e78877d933fc80107aa6f3790fd1cf927250b852d6cb53202be696b4903ed8ca839b809626aaf18050bf7e436fab7]
-    PUBLISHING_MATCH_REPO_TOKEN: ENCRYPTED[3d1230b744c6ed6c788a91bec741b769401dbcd426b18f9af8080bfeefdfc21913ca4047980c5b5b7ce823f12e7b6b19]
-    # Apple Certificates Match Passphrase
-    MATCH_PASSWORD: ENCRYPTED[db07f252234397090e3ec59152d9ec1831f5ecd0ef97d247b1dca757bbb9ef9b7c832a39bce2caf1949ccdf097e59a73]
-  osx_instance:
-    image: mojave-xcode-10.1
-  # occasionally the clock on these machines is out of sync
-  # with the actual time - this should help to verify
-  print_date_script:
-    - date
-  install_cocoapods_script:
-    - sudo gem install cocoapods --no-document
-  git_fetch_script:
-    - git clean -xfd
-    - git fetch origin
-    - git fetch origin master # To set FETCH_HEAD
-  setup_script:
-    - bin/flutter config --no-analytics
-    - bin/flutter update-packages
-  test_all_script:
-    - ./dev/bots/deploy_gallery.sh
-
-task:
-  use_compute_credits: $CIRRUS_USER_COLLABORATOR == 'true'
   osx_instance:
-    image: mojave-xcode-10.1
-  depends_on:
-    - analyze
+    image: mojave-xcode-10.2-flutter # see https://cirrus-ci.org/guide/macOS/ for list of images (we should update regularly)
```

Comment:
We should keep this in sync with the devicelab.  @digiter

Reply:
Previously we used a variety of images on macOS. This image maximized the number of shards I could run correctly while keeping a single image.

I think variety is good, actually -- it's fine if devicelab is on a different version (or has many versions, like now), so long as we are consistent from commit to commit. I just wanted the .cirrus.yml file to be more coherent in how it specified the macOS image.